plugins/keyboard-navigation/src/navigation_controller.js

```diff
@@ -851,7 +851,7 @@ export class NavigationController {
           return false;
         }
         this.navigation.moveCursorOnBlockDelete(workspace, sourceBlock);
-        Blockly.deleteBlock(sourceBlock);
+        this.navigation.deleteBlock(sourceBlock);
```

Comment:
instead of copying the deleteBlock function , you should be able to use `sourceBlock.checkAndDelete()`

well, that's the intention anyway, if that isn't working we should look into it

Reply:
Thanks! This works perfectly.